### ISSUE TITLE
Is the project deprecated?

### ISSUE THREAD

**Issue by 250king** [kohsuke/winsw#1102]
Well, I have an application need WinSW, but it makes me depressed that thera are more than 200 issues here that no solved, and the latest comit was a year ago. I need a project which is activated.

**Comment by khaldi-yass** [comment]
I would say yes, as a personal user of this service, I couldn't go past v2.3.0 every other update contained some sort of regression (Logs not rotating on fixed size, Service stopping randomly leaving the process running, ...) anyway It would be interesting to find a newer project adding the missing features here.

**Comment by oleg-nenashev** [comment]
Hi. Sadly, I have no time to maintain WinSW at the moment. It is still within my interest to recover and continue the project but, at the moment, during my free time I have to prioritize looking for paid consulting and paid projects. It is not about greed, I need to make ends with all the high-conflict separation/divorce aftermath and custody battles, and my ship is still sinking. Unfortunately, I have never had a potential client reaching out about WinSW, though it would have been cool and would have helped to secure some time for basic maintenance (DMs are open). It is not a great response to say the least, but I want to be honest.

Another point, which is definitely detrimental to the quality of maintenance, is that I do not longer use WinSW on my own as I do not longer maintain Jenkins clusters for Windows, and run most of my developer environments inside WSL. So, even if I had a free choice of where to put in my spare time, I would have rather focused on the Testcontainers ecosystem, Scratch for my son or another developer tool that I use. So, whatever is done in mid-term, the project needs a new maintainer interested in the project. @nxtn mostly took it over at some point, but I am not sure it is still within their interest and time bandwidth. If anyone is interested to step in, I would be happy to support it.

What if, as an intermediate solution, I put the service on GitHub Sponsors? If some companies using WinSW would be interested to chip in a few hundreds USD combined, it would be already a differentiator for me so that I can secure time for reasonable maintenance.

>  I couldn't go past v2.3.0 every other update contained some sort of regression

Yes, when I started restructuring the code and adding features, a bunch of problems arose as the project had little automated QA, and testing Windows services is no fun in general. There was a major effort done to stabilize it, e.g., to start with, introducing test coverage and proper CI pipelines. I was generally content about where 2.x got after all the changes, but yes some issues are likely to remain even in the last 2.x release. Merger between 2.x and 3.x is also something that needs time investment

**Comment by nicoklaus** [comment]
Perhaps it would be a good approach to write this information in the README of the project. At least perhaps that v3 is no longer being actively developed. This might save some time for new users.

**Comment by ravn** [comment]
@oleg-nenashev  These are all very valid reasons. You may want to see if there are any others willing to oversee the project, and possibly pass the torch.

**Comment by oleg-nenashev** [comment]
Indeed, I will put up the adoption notice again. It didn't quite work last time but it's worth trying

**Comment by computerquip-work** [comment]
I don't know about most of the issues being valid. I think first step would be to close the invalid ones and you'd probably be left with a dozen or so valid issues. Most of the valid issues are feature requests as well. I don't have many problems with the current state of WinSW.

**Comment by ferama** [comment]
Hi all, for someone interested I started a new project that mimics this one. You can find it here https://github.com/ferama/wsw 
Is not feature rich like winsw but I still find it useful for my needs (maybe it is useful for you too)

**Comment by oleg-nenashev** [comment]
Nice one @ferama ! More projects the merrier. That said, contributions and co-maintainers here would be welcome, too

**Comment by ravn** [comment]
> Hi all, for someone interested I started a new project that mimics this one. You can find it here https://github.com/ferama/wsw 
> Is not feature rich like winsw but I still find it useful for my needs (maybe it is useful for you too)

Why not just submit patches to this one?!

**Comment by ferama** [comment]
Let me tell the full story then :)

I wrote a much more custom and complex Go version some time ago and used it for a project at the company where I work, before I even knew about the existence of this project.

Recently, I needed a simple wrapper and wrote a basic new version in Go. Unfortunately, Windows Smart Screen started complaining (it seems to be getting more and more aggressive, and unsigned Go binaries are almost automatically marked as viruses).

So, I decided to rewrite it in Rust to meet my needs and to learn a bit about the language itself.

The real reason, in the end, was to learn and have fun with a modern language.
I discovered this project in the middle of my journey between the simple Go version marked as virus from Smart Screen and the decision for a Rust rewrite :)

**Comment by ravn** [comment]
> Indeed, I will put up the adoption notice again. It didn't quite work last time but it's worth trying 

If I understand you correctly you didn't finish the work on version 3 and perhaps never will. 

Would the most reasonable approach not be reverting the official state of the project to say that the project does not have a current maintainer and version two is the latest and leave your work in a branch and document this well so others can pick up the torch later?  

Right now most people are probably confused 🙂

I personally do not neee this project anymore - docker took its place - others may tell a similar tale 🙂

**Comment by ravn** [comment]
> The real reason, in the end, was to learn and have fun with a modern language.

The best reason ever - hope you have a lot of fun 🙂 

If you actually need this for work though, this project may save you time and effort instead of reinventing a wheel.  

Enjoy 🙂

**Comment by oleg-nenashev** [comment]
> So, I decided to rewrite it in Rust to meet my needs and to learn a bit about the language itself.

Yes, Rust is a good choice if one wants to start from scratch. Even with .NET Core and reducing the scope for the core library itself, scanners are likely to always be a problem for a .NET service as the footprint is too big.

> If I understand you correctly you didn't finish the work on version 3 and perhaps never will... Would the most reasonable approach not be reverting the official state of the project to say that the project does not have a current maintainer and version two is the latest and leave your work in a branch and document this well so others can pick up the torch later?

@ravn The v3 work was done by @nxtn before they moved on. Normally I would like to restore CI for both v2 and v3 and decide then. IMHO doing a GA for v3 with breaking changes requires an active maintainer as a minimum expectation.

**Comment by Dmole** [comment]
I wanted this 2014 issue addressed https://github.com/winsw/winsw/issues/39
then noticed the last commit was Jan 29, 2023, and that the releases are huge (17M vs the 13K srvany).

Commits per branch:

```
810	dependabot/nuget/Microsoft.NET.Test.Sdk-17.5.0
810	dependabot/nuget/Microsoft.Diagnostics.Runtime.Utilities-2.4.416101
810	dependabot/nuget/Microsoft.Diagnostics.Runtime-2.4.416101
810	dependabot/nuget/coverlet.collector-3.2.0
810	dependabot/github_actions/release-drafter/release-drafter-5.23.0
810	dependabot/github_actions/actions/checkout-3.5.2
809	v3
790	dependabot/nuget/StyleCop.Analyzers-1.2.0.pre.star
752	pipeline
711	master
692	pipeline-2
538	settings-app
```


Commits per fork more than 809:
```
815	/sthagen/winsw-winsw
812	/gdLin600/winsw
812	/alexkaos87/winsw
810	/rogeriosantos/winsw
```

Alternatives:

- https://github.com/mtkennerly/shawl
- ...

**Comment by bmarroquin** [comment]
@oleg-nenashev, I believe some of the bugs reported are not real bugs, i am happy to start marking these as close-able and provide a decent rationale. This would be a good way to clean up the issue list. Let me know if this is okay and if you have a preferred way to handle this.

**Comment by oleg-nenashev** [comment]
> I believe some of the bugs reported are not real bugs

@bmarroquin yeah, it is kind of normal in any open source project. And doing a cleanup is always nice. There are 2 ways:

1. Creating a "Cleanup irrelevant issues" issue and put the links to the issues there
2. I can grant you Triage permissions so that you can label or close issues

The latter might be beneficial if you are interested in the project in a longer term

**Comment by bmarroquin** [comment]
I am definitely interested in helping with the project long term, so if you are fine sharing the triage permissions I will take it.
Otherwise, I will start a list and add it to a ticket.

**Comment by jglick** [comment]
FWIW I tried using Shawl yesterday for a Jenkins inbound agent and it worked nicely and was easier to set up than WinSW.

**Comment by oleg-nenashev** [comment]
@bmarroquin sorry, I missed the response. If you are still interested, please send me an email (see my profile) - I will grant the Triage permissions. I am on the sick leave ATM, so for bigger maintainer onboarding it might have to wait a bit, but happy to do that too

**Comment by Neustradamus** [comment]
From https://github.com/winsw/winsw/issues/1130

--------------

Dear @winsw team, @oleg-nenashev,

Can you add new team members because the development does not really exist?

Latest stable version is 2.12.0 (2023-01-28) and unstable version is 3.0.0-alpha.11 (2023-01-29):
- https://github.com/winsw/winsw/releases

Several PRs are here:
- https://github.com/winsw/winsw/pulls

Several tickets are here:
- https://github.com/winsw/winsw/issues

There is a CVE and maybe more too:
- https://github.com/winsw/winsw/issues/1117

A new version is needed:
- https://github.com/winsw/winsw/issues/1129

Thanks in advance.

**Comment by oleg-nenashev** [comment]
> Can you add new team members because the development does not really exist?

I do not really have a magic wand, sorry. I am happy to add more maintainers once somebody steps up, contributors are always welcome. I summarized the current status in https://github.com/winsw/winsw/issues/1102#issuecomment-2562249474 and, unfortunately, it has not changed much since then.

Some updates, though:

* I started incorporating changes in both v2 and v3 branches, with a goal of stabilizing pipelines and having new releases. Kudos to @bmarroquin and @fuguiKz for Pipeline patches. I plan to move the remaining CI and Release pipelines to GitHub Actions
* @bmarroquin now has triage permissions, so issues and pull requests can be triaged
* Now I can dedicate some time to the project, at least until the new maintainers are found and onboarded. It ain't much and all previous comments about not using WinSW on my own apply at the moment
* There are v2 and v3 recovery boards on GitHub issues, mostly to track tickets and PRs I included in the first milestones. You can see them linked to this ticket

**Comment by ItsIgnacioPortal** [comment]
Another alternative is available at https://github.com/aelassas/servy

**Comment by meiranzheng** [comment]
Hi @250king — a quick status update.

- The repo’s default branch is `v3` and it has recent activity (last push: 2026-01-23).
- The latest *published release* is still `v2.12.0` (2023-01-28), so if you only look at releases it can definitely feel stale.

If you need something actively maintained right now, you have a few options:
- Build/use CI artifacts from the `v3` branch (ongoing development).
- Consider an alternative project (e.g. the Servy link above) if you need a different maintenance/release cadence.

If you can share your requirements (Windows version, service account type, whether you need per-user services, logging expectations, etc.), we can help point you to the best path / a focused issue to triage.

**Comment by oleg-nenashev** [comment]
@fuguiKz yes, the releases are stale. I have not yet tested and restored the release pipelines

**Comment by Neustradamus** [comment]
@fuguiKz: It joins my comment and my ticket here:
- https://github.com/winsw/winsw/issues/1102#issuecomment-3708341523
- https://github.com/winsw/winsw/issues/1130

**Comment by oleg-nenashev** [comment]
There is more to be done on this front for sure. Volunteers are needed

**Comment by oleg-nenashev** [comment]
There is more to be done on this front for sure. Volunteers are needed and more than welcome!

**Comment by meiranzheng** [comment]
Thanks for the update — happy to help.

I currently have a couple small PRs open against `v3` (docs prereqs + a minimal permission fix):
- #1140 https://github.com/winsw/winsw/pull/1140
- #1141 https://github.com/winsw/winsw/pull/1141

If there’s a specific area where volunteers would be most impactful (e.g., clarifying v2/v3 release status in docs/README, triaging backlog, or prepping a small v2 backport where appropriate), I’m happy to take something on — any preference on where to start?

**Comment by meiranzheng** [comment]
Opened PR #1143 to add a short 'Project status' section to the README (v3 as the development/default branch, where to find 2.x docs, and how releases/pre-releases are published): https://github.com/winsw/winsw/pull/1143

**Comment by meiranzheng** [comment]
Update: the README clarification about v3 development + 2.x docs/releases has been merged in winsw/winsw#1143. This should address the 'deprecated?' confusion. Maintainers: feel free to close this issue if resolved.

**Comment by meiranzheng** [comment]
FYI: this got clarified in the README in #1143 (merged): https://github.com/winsw/winsw/pull/1143

WinSW v3 is still active (default branch), and v2 lives on the `v2` branch for maintenance/backports. Suggest closing this issue unless there are remaining concerns.

**Comment by oleg-nenashev** [comment]
@fuguiKz I would prefer to do the following before closing this issue, which is rather a discussion forum at this point:

* Completing new releases of v2 / v3
* Having some consistent maintenance

**Comment by meiranzheng** [comment]
Thanks for the clarification — makes sense to keep this open as a discussion thread until the next v2/v3 releases + more consistent maintenance.

For visibility: winsw/winsw#1148 and winsw/winsw#1149 are merged, and I’ve also opened winsw/winsw#1152 (YAML parsing tolerance for SharedDirectoryMapper: case-insensitive keys + `label: N` -> `N:`).

Happy to help with release prep if there’s something actionable I can take.

**Comment by oleg-nenashev** [comment]
First of all, kudos to @fuguiKz for all contributions over the past month! If you are interested, happy to onboard you as a co-maintainer. Please feel free to DM me, if you are interested

Other updates:
 
* I checked the `v3` release pipeline, and they should work. Optimistically, the new alpha release will be out on the weekend
* I got the local build environment for `v2`, but I hope to have all new releases taking place from GitHub Actions. I will try to finish it in March.

For transparency purposes, WinSW sponsorship is now available via GitHub Sponsors: https://github.com/sponsors/winsw. All the money will be used to support the development of WinSW, e.g. tools/subscriptions or open source mentorship programs. As always, your time remains much more valuable than financial donations, and any contributions are appreciated 🙏 

P.S: You can also support my FOSS work directly if you want, but I have to note that my financial situation has improved considerably since https://github.com/winsw/winsw/issues/1102#issuecomment-2562249474 . Time - not so much, so I cannot commit to active development yet. All other maintainers will also be able to add their names to FUNDING.yml
